Libplanet.Net.Tests/Consensus/GossipTest.cs

```diff
@@ -40,19 +40,74 @@ public void Dispose()
             NetMQConfig.Cleanup();
         }
 
+        [Fact(Timeout = Timeout)]
+        public async void PublishMessage()
+        {
+            MemoryStoreFixture fx = new MemoryStoreFixture();
+            bool received1 = false;
+            bool received2 = false;
+            var key1 = new PrivateKey();
+            var key2 = new PrivateKey();
+            var receivedEvent = new AsyncAutoResetEvent();
+            var gossip1 = CreateGossip(
+                content =>
+                {
+                    if (content is ConsensusProposalMsg)
+                    {
+                        received1 = true;
+                    }
+                },
+                key1,
+                6001,
+                new[] { new BoundPeer(key2.PublicKey, new DnsEndPoint("127.0.0.1", 6002)) });
+            var gossip2 = CreateGossip(
+                content =>
+                {
+                    if (content is ConsensusProposalMsg)
+                    {
+                        received2 = true;
+                        receivedEvent.Set();
+                    }
+                },
+                key2,
+                6002,
+                new[] { new BoundPeer(key1.PublicKey, new DnsEndPoint("127.0.0.1", 6001)) });
+            try
+            {
+                _ = gossip1.StartAsync(default);
+                _ = gossip2.StartAsync(default);
+                await gossip1.WaitForRunningAsync();
+                await gossip2.WaitForRunningAsync();
+                gossip1.PublishMessage(
+                    TestUtils.CreateConsensusPropose(fx.Block1, new PrivateKey(), 0));
+                await receivedEvent.WaitAsync();
+                Assert.True(received1);
+                Assert.True(received2);
```

Comment:
I think this should test somehow how fast the message is received. Maybe put a cap on using a timer?

Reply:
I agree, but it is not easy to implement because broadcast also has its own delay and it is also random.

IMO, to check whether the message is received via `HaveMsg` is most accurate, but it needs some "holes"...

Reply:
I was thinking something like the following:

```csharp
                Stopwatch stopwatch = new Stopwatch();
                var proposeMsg = TestUtils.CreateConsensusPropose(fx.Block1, new PrivateKey(), 0);
                _ = gossip1.StartAsync(default);
                _ = gossip2.StartAsync(default);
                await gossip1.WaitForRunningAsync();
                await gossip2.WaitForRunningAsync();
                stopwatch.Start();
                gossip1.PublishMessage(proposeMsg);
                await receivedEvent.WaitAsync();
                stopwatch.Stop();
                Assert.True(stopwatch.ElapsedMilliseconds < 100);
                Assert.True(received1);
                Assert.True(received2);
```

This is rather hacky tho. 🙄